src/interpreter.rs

```diff
@@ -137,7 +137,11 @@ impl<'a, 'tcx> GlobalEvalContext<'a, 'tcx> {
             tcx: tcx,
             mir_map: mir_map,
             mir_cache: RefCell::new(DefIdMap()),
-            memory: Memory::new(),
+            memory: Memory::new(tcx.sess
```

Comment:
In the future, it would be best if `Memory` had access to `tcx.data_layout` (the `TargetDataLayout`). This would also allow it to use primitive type alignments to diagnose unaligned memory accesses.

But you don't need to make that change now. I just wanted to make a note of it.